components/devtools/actors/console.rs

```diff
@@ -191,76 +290,34 @@ impl Actor for ConsoleActor {
-                    BooleanValue(val) => Value::Bool(val),
-                    NumberValue(val) => {
-                        if val.is_nan() {
-                            let mut m = Map::new();
-                            m.insert("type".to_owned(), Value::String("NaN".to_owned()));
-                            Value::Object(m)
-                        } else if val.is_infinite() {
-                            let mut m = Map::new();
-                            if val < 0. {
-                                m.insert("type".to_owned(), Value::String("-Infinity".to_owned()));
-                            } else {
-                                m.insert("type".to_owned(), Value::String("Infinity".to_owned()));
-                            }
-                            Value::Object(m)
-                        } else if val == 0. && val.is_sign_negative() {
-                            let mut m = Map::new();
-                            m.insert("type".to_owned(), Value::String("-0".to_owned()));
-                            Value::Object(m)
-                        } else {
-                            Value::Number(Number::from_f64(val).unwrap())
-                        }
-                    },
-                    StringValue(s) => Value::String(s),
-                    ActorValue { class, uuid } => {
-                        //TODO: make initial ActorValue message include these properties?
-                        let mut m = Map::new();
-                        let actor = ObjectActor::new(registry, uuid);
+                let msg = self.evaluateJS(&registry, &msg);
+                stream.write_json_packet(&msg);
+                ActorMessageStatus::Processed
+            },
 
-                        m.insert("type".to_owned(), Value::String("object".to_owned()));
-                        m.insert("class".to_owned(), Value::String(class));
-                        m.insert("actor".to_owned(), Value::String(actor));
-                        m.insert("extensible".to_owned(), Value::Bool(true));
-                        m.insert("frozen".to_owned(), Value::Bool(false));
-                        m.insert("sealed".to_owned(), Value::Bool(false));
-                        Value::Object(m)
-                    },
+            "evaluateJSAsync" => {
+                //TODO: use a timestamp for resultID
+                let resultID = 0;
+                let early_reply = EvaluateJSAsyncReply {
+                    from: self.name(),
+                    resultID: resultID,
                 };
-
-                //TODO: catch and return exception values from JS evaluation
-                let msg = EvaluateJSReply {
+                // Emit an eager reply so that the client starts listening
+                // for an async event with the resultID
+                stream.write_json_packet(&early_reply);
+                let reply = self.evaluateJS(&registry, &msg).unwrap();
+                let msg = EvaluateJSEvent {
```

Comment:
I hope there is a better way to do this..

Reply:
You could have an intermediate type with `From` impls but I don't think it's a big deal.